mne/label.py

```diff
@@ -1659,6 +1660,159 @@ def _grow_nonoverlapping_labels(subject, seeds_, extents_, hemis, vertices_,
+        rest = len(parc) - 1
+        # grow from sources
+        while rest:
+            if not edge:
+                rest_idx = np.where(parc < 0)[0]
+                s = random.choice(rest_idx)
+                edge = [s]
+                label_idx += 1
+                label_size = 1
+                parc[s] = label_idx
+                rest -= 1
+
+            vert_from = edge.pop(0)
+
+            # add neighbors within allowable distance
+            row = graph[vert_from, :]
+            for vert_to, dist in zip(row.indices, row.data):
+                vert_to_label = parc[vert_to]
+
+                # abort if the vertex is already occupied
+                if vert_to_label >= 0:
+                    continue
+
+                # abort if outside of extent
+                if label_size > parcel_size:
+                    label_idx += 1
+                    label_size = 1
+                    edge = [vert_to]
+                    parc[vert_to] = label_idx
+                    rest -= 1
+                    break
+
+                # assign label value
+                parc[vert_to] = label_idx
+                label_size += 1
+                edge.append(vert_to)
+                rest -= 1
+
+        # merging small labels
+        # label connectivity matrix
+        n_labels = label_idx + 1
+        label_sizes = np.empty(n_labels, dtype=int)
+        label_conn = np.zeros([n_labels, n_labels], dtype='bool')
+        for i in range(n_labels):
+            vertices = np.nonzero(parc == i)[0]
+            label_sizes[i] = len(vertices)
+            neighbor_vertices = graph[vertices, :].indices
+            neighbor_labels = np.unique(np.array(parc[neighbor_vertices]))
+            label_conn[i, neighbor_labels] = 1
+        np.fill_diagonal(label_conn, 0)
+
+        # merging
+        label_id = range(n_labels)
+        while n_labels > n_parcel / 2:
+            # smallest label and its smallest neighbor
+            i = np.argmin(label_sizes)
+            neighbors = np.nonzero(label_conn[i, :])[0]
+            j = neighbors[np.argmin(label_sizes[neighbors])]
+
+            # merging two labels
+            label_conn[j, :] += label_conn[i, :]
+            label_conn[:, j] += label_conn[:, i]
+            label_conn = np.delete(label_conn, i, 0)
+            label_conn = np.delete(label_conn, i, 1)
+            label_conn[j, j] = 0
+            label_sizes[j] += label_sizes[i]
+            label_sizes = np.delete(label_sizes, i, 0)
+            n_labels -= 1
+            vertices = np.nonzero(parc == label_id[i])[0]
+            parc[vertices] = label_id[j]
+            label_id = np.delete(label_id, i, 0)
+
+        # convert parc to labels
+        for i in xrange(n_labels):
+            vertices = np.nonzero(parc == label_id[i])[0]
+            name = 'label_' + str(i)
+            label_ = Label(vertices, hemi=hemi, name=name, subject=subject)
+            labels.append(label_)
+        parc_both += [parc]
```

Comment:
is this the same algo as in brainstorm?

Reply:
I followed the same logic as brainsotm code:
 - define number of vertices per parcel
 - grow parcels till there are no free neighbor vertices or till riches max size
 - merge smallest parcels till rich the needed number